### PR TITLE
[BUTTON] Dropdown

### PR DESCRIPTION
**Pull request by mikorosales**
## Issue Link
[LMS-325](https://framgiaph.backlog.com/view/LMS-325)

## Definition of Done
A dropdown component should be created under shared > components

**For your convenience you can access this link**
- http://localhost:3000/demo/components#dropdown

## Steps to reproduce
import the dropdown component in the page you want it to be utilized
```
import Dropdown from '@/src/shared/components/Dropdown';


<Dropdown />
```
## Affected Components / Functionalities / Page
n/a

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- Dropdown with 3 options can be seen in http://localhost:3000/demo/components
- A dropdown component has been created under shared > components.

Comments: 0
Test time: 15 mins
Iterations: 0

File: client/src/pages/demo/components/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { Fragment } from 'react';
+import Dropdown from '@/src/shared/components/Dropdown';
+
+const Components: React.FC = () => {
+  return (
+    <Fragment>
+      <div className="p-4 border-b border-black">
+        <Dropdown />
+        <div className="mt-[5px]">
+          <h1>Props: </h1>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:10]: - please add anchor tag para dali ra e-navigate.
```suggestion
      <div className="p-4 border-b border-black">
        <a href="#dropdown">Dropdown</a>
        <Dropdown />
        <div className="mt-[5px]">
          <h1>Props: </h1>
```

File: client/src/pages/demo/components/index.tsx [hunk 2]
```diff
@@ -32,6 +33,23 @@ const DemoComponent: React.FunctionComponent = () => {
         <br />
         <hr className="w-3/4 mx-auto" />
       </div>
+
+      <div className="p-4 border-b border-black">
+        <Dropdown>
+          <a href="#dropdown">Dropdown</a>
+        </Dropdown>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:40]: - i don't understand why you put anchor tag inside dropdown component.
- you supposed to demonstrate the dropdown component and its props :bow:

File: client/src/pages/demo/components/index.tsx [hunk 3]
```diff
@@ -32,6 +33,23 @@ const DemoComponent: React.FunctionComponent = () => {
...
+        <div className="mt-[5px]">
+          <h1>Props: Dropdown</h1>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:42]: - while this one should be an anchor tag.

File: client/src/pages/demo/components/index.tsx [hunk 4]
```diff
@@ -32,6 +33,24 @@ const DemoComponent: React.FunctionComponent = () => {
         <br />
         <hr className="w-3/4 mx-auto" />
       </div>
+
+      <div className="p-4 border-b border-black">
+        <a id="dropdown"></a>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:38]: - change `id` to `href`

File: client/src/pages/demo/components/index.tsx [hunk 5]
```diff
@@ -32,6 +33,23 @@ const DemoComponent: React.FunctionComponent = () => {
         <br />
         <hr className="w-3/4 mx-auto" />
       </div>
+
+      <div className="p-4 border-b border-black" id='dropdown'>
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:37]: - as per standard. use double quotes for html tags
```suggestion
      <div className="p-4 border-b border-black" id="dropdown">
```

**Review by Marielleregner7 — COMMENTED** [review]
**Comment**
- Please fix lint errors 🙇

image

File: client/src/shared/components/Dropdown/index.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            ))}
+          </div>
+        </div>
+      )}
+    </div>
+  );
+};
+
+Dropdown.defaultProps = {
+  options: ['option1', 'option2', 'option3'],
```
Comment (escbooster12-dev) [inline client/src/shared/components/Dropdown/index.tsx:77]: - `options` prop is required. so do not specify default value.

File: client/src/shared/components/Dropdown/index.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  label: 'Dropdown menu'
```
Comment (escbooster12-dev) [inline client/src/shared/components/Dropdown/index.tsx:78]: - while `label` should be optional. and should have just an empty string as default value

File: client/src/shared/icons/ChevronDown.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const ChevronDown: React.FC = () => {
```
Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronDown.tsx:3]: - add necessary props (refer to existing icons)

File: client/src/shared/components/Dropdown/index.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+
+import ChevronDown from '@/src/shared/icons/ChevronDown';
+
+interface DropdownProps {
+  options: string[]
+  label: string
+}
```
Comment (escbooster12-dev) [inline client/src/shared/components/Dropdown/index.tsx:8]: - this interface could be usable in the future. Please do export it.
```suggestion
export interface DropdownProps {
  options: string[]
  label: string
}
```

**could you apply this as well to all existing components**

File: client/src/shared/components/Dropdown/index.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+import ChevronDown from '@/src/shared/icons/ChevronDown';
+
+export interface Option {
+  text: string;
+  url: string;
+}
+
+export interface DropdownProps {
+  label: string;
```
Comment (escbooster12-dev) [inline client/src/shared/components/Dropdown/index.tsx:11]: - optional prop
```suggestion
  label?: string;
```

File: client/src/shared/icons/ChevronDown.tsx [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
```
Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronDown.tsx:2]: - as per standard please change file name `ChevronDownIcon`

i noticed some existing icons does not have `postfix` (`Icon`). Could apply it as well :bow:

File: client/src/shared/icons/ChevronDown.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+interface ChevronDownIconProps {
+  height: number;
+  width: number;
```
Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronDown.tsx:5]: - add `className` and `color`

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- Dropdown with 3 options can be seen in http://localhost:3000/demo/components#dropdown
- A dropdown component has been created under shared > components.
- No ESLint warnings or errors.

Comments: 15
Test time: 30 mins
Iterations: 2

File: client/src/shared/icons/ChevronDownIcon.tsx [hunk 13]
```diff
@@ -1,13 +1,17 @@
...
+  className: string
+  color: string
 }
 
 const ChevronDown: React.FC<ChevronDownIconProps> = ({
   height,
-  width
+  width,
+  className,
+  color
```
Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronDownIcon.tsx:14]: - set as optional and set default value for it.

Comment (escbooster12-dev) [inline client/src/shared/icons/ChevronDownIcon.tsx:14]: - where is the application? i did not see you apply it in the icon's svg

File: client/src/shared/layouts/layout-learning.tsx [hunk 14]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import NextIcon from '../icons/next';
-import PreviousIcon from '../icons/previous';
+import NextIcon from '../icons/nextIcon';
```
Comment (escbooster12-dev) [inline client/src/shared/layouts/layout-learning.tsx:2]: - for shared components file naming. use _Pascal case_ as per standard
image

**apply to all related**

File: client/src/shared/components/Dropdown/index.tsx [hunk 15]
```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+
+import ChevronDown from '@/src/shared/icons/chevronDownIcon';
```
Comment (escbooster12-dev) [inline client/src/shared/components/Dropdown/index.tsx:3]:
```suggestion
import ChevronDownIcon from '@/src/shared/icons/chevronDownIcon';
```

File: client/src/shared/layouts/layout-learning.tsx (hunk 14 above)
Comment (mikorosales) [reply client/src/shared/layouts/layout-learning.tsx]: I encountered a problem sir, I already committed the file rename changes but it does not reflect in the file changes, but in my local machine it is reflected

image

File: client/src/shared/layouts/LayoutLearning.tsx [hunk 16]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
-import NextIcon from '../icons/next';
-import PreviousIcon from '../icons/previous';
-import Sidebar from './sidebar';
+import NextIcon from '../icons/NextIcon';
+import PreviousIcon from '../icons/PreviousIcon';
+import Sidebar from './Sidebar';
```
Comment (kdacudag) [inline client/src/shared/layouts/LayoutLearning.tsx:4]: Always use module path aliases
```suggestion
import NextIcon from '@/src/shared/icons/NextIcon';
import PreviousIcon from '@/src/shared/icons/PreviousIcon';
import Sidebar from '@/src/shared/layouts/Sidebar';
```

File: client/src/shared/layouts/Sidebar.tsx [hunk 17]
```diff
@@ -1,7 +1,7 @@
 import Link from 'next/link';
 import React from 'react';
-import MenuLogo from '../icons/menu';
-import PreviousIcon from '../icons/previous';
+import MenuLogo from '../icons/MenuIcon';
+import PreviousIcon from '../icons/PreviousIcon';
```
Comment (kdacudag) [inline client/src/shared/layouts/Sidebar.tsx:4]: Always use module path aliases
```suggestion
import MenuLogo from '@/src/shared/icons/MenuIcon';
import PreviousIcon from '@/src/shared/icons/PreviousIcon';
```

**Review by Marielleregner7 — COMMENTED** [review]
Comments: 23
Test time: 30 mins
Iterations: 9

**Review by kdacudag — APPROVED** [review]
Approved

**Pull request merged**